### PR TITLE
[eas-cli] proxy support

### PR DESCRIPTION
**Pull request by wkozyra95**
# Checklist

- [x] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary. You can comment this pull request with `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]` and CHANGELOG.md will be updated automatically.

# Why

Add proxy support

# How

proxy traffic if `http_proxy` env is present

# Test Plan

Setup local proxy and verify all requests go through it ( tested with `eas credentials` including apple authentication)

### CODE REVIEW

File: packages/eas-cli/src/fetch.ts [hunk 1]
```diff
@@ -8,6 +10,16 @@ export class RequestError extends Error {
   }
 }
 
+function createHttpAgent(): Agent | null {
+  const httpProxyUrl = process.env.http_proxy;
+  if (!httpProxyUrl) {
+    return null;
+  }
+  return createHttpsProxyAgent(httpProxyUrl);
```
Comment (dsokal) [inline packages/eas-cli/src/fetch.ts:18]: Does the package validate the URL? If not, maybe it's worth adding custom validation.

File: packages/eas-cli/src/fetch.ts [hunk 2]
```diff
@@ -8,6 +10,16 @@ export class RequestError extends Error {
   }
 }
 
+function createHttpAgent(): Agent | null {
+  const httpProxyUrl = process.env.http_proxy;
```
Comment (dsokal) [inline packages/eas-cli/src/fetch.ts:14]: Is it some kind of standard that this is in lowercase (`http_proxy`)?

File: packages/eas-cli/src/fetch.ts [hunk 3]
```diff
@@ -8,6 +10,16 @@ export class RequestError extends Error {
...
+  if (!httpProxyUrl) {
+    return null;
+  }
+  return createHttpsProxyAgent(httpProxyUrl);
+}
+
+export const maybeHttpAgent: Agent | null = createHttpAgent();
```
Comment (dsokal) [inline packages/eas-cli/src/fetch.ts:21]: Maybe this should be `httpProxy`? I can see that `fetch` expects this to be called "agent" but I think it'd be better to treat it as a `fetch` "thing".

File: packages/eas-cli/src/fetch.ts (hunk 1 above)
Comment (wkozyra95) [reply packages/eas-cli/src/fetch.ts:18]: you will get error like this `[Network] request to https://api.expo.dev/graphql failed, reason: connect ENOENT lksdjflksdjf`

also value here can have a formal `http://username:password@example.com/` not sure if this is valid url

File: packages/eas-cli/src/fetch.ts (hunk 2 above)
Comment (wkozyra95) [reply packages/eas-cli/src/fetch.ts:14]: yes, but upper case also is a valid option, in our docs we are listing multiple value that should be set. npm rc is using lowercase value so i decided to go with that here

https://docs.expo.dev/guides/troubleshooting-proxies/#for-command-line-applications

File: packages/eas-cli/src/fetch.ts (hunk 3 above)
Comment (wkozyra95) [reply packages/eas-cli/src/fetch.ts:21]: this is http agent, the fact that it is changing the request to connect to proxy is not relevant to HTTP clients

**Review by dsokal — APPROVED** [review]

**Pull request merged**